src/avram/connection.cr

```diff
@@ -1,10 +1,17 @@
 # Handles the connection to the DB.
 class Avram::Connection
+  private getter db : DB::Database? = nil
+
   def initialize(@connection_string : String, @database_class : Avram::Database.class)
   end
 
   def open : DB::Database
-    try_connection!
+    @db = try_connection!
+  end
+
+  def close
+    @db.try(&.close)
+    @db = nil
   end
```

Comment:
I don't really understand this. We've never had a `db.close` before, but it feels like there should have been... I think we just open a connection, then let it sit open in the connection pool forever. If you redeploy your app, that kills all of the connections which may explain why we always get IO errors randomly... But what's a better way to handle this?